src/components/molecules/Schedule/Schedule.tsx

```diff
@@ -0,0 +1,183 @@
+            (acc, event) => Math.min(acc, getHours(eventStartTime(event))),
+            SCHEDULE_MAX_START_HOUR
+          )
+        ),
+        SCHEDULE_MAX_START_HOUR
+      ),
+    [locatedEvents]
+  );
+
+  const scheduleStartDateTime = useMemo(
+    () => setHours(scheduleDate, scheduleStartHour),
+    [scheduleStartHour, scheduleDate]
+  );
+
+  // pairs (venueId, roomTitle) are unique because they are grouped earlier (see NavBarSchedule#schedule)
+  const roomCells = useMemo(
+    () =>
+      locatedEvents?.map(({ location, events }) => (
+        <div
+          key={`RoomCell-${location.venueId}-${location.roomTitle}`}
+          className="Schedule__room"
+        >
+          <p className="Schedule__room-title">
+            {location.roomTitle || location.venueTitle || location.venueId}
+          </p>
+          <span className="Schedule__events-count">
+            {formatMeasurement(events.length, "event")}
+          </span>
+        </div>
+      )),
+    [locatedEvents]
+  );
+
+  const hoursRow = useMemo(
+    () =>
+      eachHourOfInterval({
+        start: scheduleStartDateTime,
+        end: endOfDay(scheduleStartDateTime),
+      }).map((scheduleHour) => (
+        <span className="Schedule__hour" key={scheduleHour.toISOString()}>
+          {format(scheduleHour, "h a")}
+        </span>
+      )),
+    [scheduleStartDateTime]
+  );
+
+  const [currentTimePosition, setCurrentTimePosition] = useState(0);
+
+  const calcCurrentTimePosition = useCallback(
+    () =>
+      setCurrentTimePosition(
+        calcStartPosition(getUnixTime(Date.now()), scheduleStartHour)
+      ),
+    [scheduleStartHour]
+  );
+
+  useEffect(() => calcCurrentTimePosition(), [calcCurrentTimePosition]);
+
+  useInterval(() => {
+    calcCurrentTimePosition();
+  }, SCHEDULE_CURRENT_TIMELINE_MS);
+
+  const containerVars = useCss({
+    "--room-count": locatedEvents.length + 1, // +1 is needed for the 1st personalized line of the schedule
+    "--current-time--position": currentTimePosition,
+    "--hours-count": hoursRow.length,
+    "--hour-width": `${SCHEDULE_HOUR_COLUMN_WIDTH_PX}px`,
+  });
+
+  const containerClasses = classNames("Schedule", containerVars);
+
+  // pairs (venueId, roomTitle) are unique because they are grouped earlier (see NavBarSchedule#schedule)
+  const rowsWithTheEvents = useMemo(
+    () =>
+      locatedEvents.map(({ location, events }) => (
+        <ScheduleRoomEvents
+          key={`ScheduleRoomEvents-${location.venueId}-${location.roomTitle}`}
+          events={events}
+          scheduleStartHour={scheduleStartHour}
+        />
+      )),
+    [locatedEvents, scheduleStartHour]
+  );
+
+  if (isLoading)
+    return (
+      <div className={containerClasses}>
+        <Loading message={"Events are loading"} />
+      </div>
+    );
+
+  if (!hasEvents)
+    return (
+      <div className={containerClasses}>
+        <div className="Schedule__no-events">No events scheduled</div>
+      </div>
+    );
```

Comment:
These if blocks should be wrapped in `{` `}` since they are multi-line.

Also, do we need to wrap `<Loading` in the `div` with `containerClasses`?

Reply:
wrapped: https://github.com/sparkletown/sparkle/pull/1426/commits/880f966fdbeb06065f5946456902e13410b88582

I use `containerClasses` here for padding-top. Or do you think it's a better idea to add a class to `<Loading />` ?

Reply:
added a `containerClassName` prop: https://github.com/sparkletown/sparkle/pull/1426/commits/5112f98528054f6eaffb0bf21a786d69cf0090d2